python/pyspark/sql/connect/dataframe.py

```diff
@@ -1685,8 +1710,12 @@ def mapInArrow(
 
     mapInArrow.__doc__ = PySparkDataFrame.mapInArrow.__doc__
 
-    def writeStream(self, *args: Any, **kwargs: Any) -> None:
-        raise NotImplementedError("writeStream() is not implemented.")
+    @property
+    def writeStream(self) -> DataStreamWriter:
+        assert self._plan is not None
+        return DataStreamWriter(plan=self._plan, session=self._session)
+
+    writeStream.__doc__ = PySparkDataFrame.writeStream.__doc__
```

Comment:
Hm, this seems quite flaky:

```
File "/__w/spark/spark/python/pyspark/sql/connect/dataframe.py", line ?, in pyspark.sql.connect.dataframe.DataFrame.writeStream
Failed example:
    with tempfile.TemporaryDirectory() as d:
        # Create a table with Rate source.
        df.writeStream.toTable(
            "my_table", checkpointLocation=d)
Exception raised:
    Traceback (most recent call last):
      File "/usr/lib/python3.9/doctest.py", line 1336, in __run
        exec(compile(example.source, filename, "single",
      File "<doctest pyspark.sql.connect.dataframe.DataFrame.writeStream[3]>", line 3, in <module>
        df.writeStream.toTable(
      File "/usr/lib/python3.9/tempfile.py", line 965, in __exit__
        self.cleanup()
      File "/usr/lib/python3.9/tempfile.py", line 969, in cleanup
        self._rmtree(self.name)
      File "/usr/lib/python3.9/tempfile.py", line 951, in _rmtree
        _rmtree(name, onerror=onerror)
      File "/usr/lib/python3.9/shutil.py", line 722, in rmtree
        onerror(os.rmdir, path, sys.exc_info())
      File "/usr/lib/python3.9/shutil.py", line 720, in rmtree
        os.rmdir(path)
    OSError: [Errno 39] Directory not empty: '/__w/spark/spark/python/target/b03d461e-84a3-4dfe-bf25-4d813d4da3a3/tmp48azcvu_'
**********************************************************************
   1 of   4 in pyspark.sql.connect.dataframe.DataFrame.writeStream
```

https://github.com/apache/spark/actions/runs/4858380330/jobs/8659766804